File: Swift/BasicExample/ATM11399/ATM11399/main.swift
//
//  main.swift
//  ATM11399
//
//  Created by 조윤영 on 2020/12/27.
//

import Foundation

var N: Int = 0
var times: [Int] = []

public func input() {
    let N = Int(readLine()!)
    times = readLine()!.split(separator: " ").map { Int($0)! }
}

public func  solution() -> Int {
    times.sort()
    var result: Int = 0
    var answer: Int = 0
    for t in times {
        result += t
        answer += result
    }
    return answer
}

input()
print(solution())

File: C++/BasicExample/ATM11399/ATM11399/main.cpp
//
//  main.cpp
//  ATM11399
//
//  Created by 조윤영 on 2020/12/26.
//

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int N;
vector<int> times;
void input() {
    cin>> N;
    times.assign(N, 0);
    for(int i=0; i<N; i++) cin>> times[i];
}

int solution() {
    int answer = 0;
    int result = 0;
    sort(times.begin(), times.end());
    for(int t: times) {
        result += t;
        answer += result;
    }
    
    return answer;
}

int main(int argc, const char * argv[]) {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    input();
    cout<<solution()<<"\n";
    return 0;
}
